src/llama-sampling.cpp

```diff
@@ -144,11 +144,12 @@ static int llama_sample_dist(llama_token_data_array * cur_p, std::mt19937 & rng)
 
         const llama_token_data * data;
 
-        bool operator==(const probs_iterator & other) const { return data == other.data; }
-        bool operator!=(const probs_iterator & other) const { return data != other.data; }
+        bool operator==(const probs_iterator other) const { return data == other.data; }
+        bool operator!=(const probs_iterator other) const { return data != other.data; }
         const float & operator*() const { return data->p; }
         probs_iterator & operator++() { ++data; return *this; }
-        probs_iterator operator++(int) { probs_iterator tmp = *this; ++data; return tmp; }
+        probs_iterator operator++(int) {
+            const probs_iterator tmp = *this; ++data; return tmp; }
```

Comment:
Remove this change

Reply:
fix it https://github.com/ggerganov/llama.cpp/pull/11294/commits/44ec40a43a3b6302a50aa934d8148cb8f5c11b71